autogluon/utils/tabular/features/generators/category.py

```diff
@@ -0,0 +1,125 @@
+import copy
+import logging
+
+from pandas import DataFrame, Series
+from pandas.api.types import CategoricalDtype
+
+from .abstract import AbstractFeatureGenerator
+from .memory_minimize import CategoryMemoryMinimizeFeatureGenerator
+from ..feature_metadata import R_CATEGORY, R_OBJECT, S_DATETIME_AS_OBJECT, S_TEXT, S_TEXT_AS_CATEGORY
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: Add hashing trick if minimize_memory=True to avoid storing full original mapping
+# TODO: Retroactively prune mappings if feature was removed downstream
+# TODO: Have a concept of a 1:1 mapping, so you can safely remove features from features_in and features_out
+#  Have a method remove_features() where each generator implements custom logic, default is just to only remove from features_out, but Category could remove from features_in + inner generators + category_map.
+class CategoryFeatureGenerator(AbstractFeatureGenerator):
+    """
+    CategoryFeatureGenerator is used to convert object types to category types, as well as remove rare categories and optimize memory usage.
+
+    Parameters
+    ----------
+    stateful_categories : bool, default True
+        If True, categories from training are applied to transformed data, and any unknown categories from input data will be treated as missing values.
+        It is recommended to keep this value as True to avoid strange downstream behaviour.
+    minimize_memory : bool, default True
+        If True, minimizes category memory usage by converting all category values to sequential integers.
+        This replaces any string data present in the categories but does not alter the behavior of models when using the category as a feature so long as the original string values are not required downstream.
+        It is recommended to keep this value as True to dramatically reduce memory usage with no cost to accuracy.
+    cat_order : str, default 'original'
+        Determines the order in which categories are stored.
+        This is important when minimize_memory is True, as the order will determine which categories are converted to which integer values.
+        Valid values:
+            'original' : Keep the original order. If the feature was originally an object, this is equivalent to 'alphanumeric'.
+            'alphanumeric' : Sort the categories alphanumerically.
+            'count' : Sort the categories by frequency (Least frequent in front with code of 0)
+    minimum_cat_count : int, default None
+        The minimum number of occurrences a category must have in the training data to avoid being considered a rare category.
+        Rare categories are removed and treated as missing values.
+        If None, no minimum count is required. This includes categories that never occur in the data but are present in the category object as possible categories.
+    maximum_num_cat : int, default None
```

Comment:
How can user specify args like `maximum_num_cat`, `minimum_cat_count` thru `TabularPrediction.fit()`

Reply:
See response to earlier question on AutoMLFeatureGenerator arguments.